app/models/deploy.rb

```diff
@@ -130,19 +130,43 @@ def self.expired
   end
 
   def buddy_name
-    user.id == buddy_id ? "bypassed" : buddy.try(:name)
+    user.id == buddy_id ? "bypassed" : buddy_with_deleted.try(:name)
   end
 
   def buddy_email
-    user.id == buddy_id ? "bypassed" : buddy.try(:email)
+    user.id == buddy_id ? "bypassed" : buddy_with_deleted.try(:email)
   end
 
   def url
     AppRoutes.url_helpers.project_deploy_url(project, self)
   end
 
+  def to_csv
+    [
+      id, project_with_deleted.name, summary, commit, job.status, updated_at,
+      start_time, user_with_deleted.try(:name), user_with_deleted.try(:email), buddy_name, buddy_email,
+      stage_with_deleted.production, stage_with_deleted.no_code_deployed, project_with_deleted.deleted_at
+    ]
+  end
+
   private
 
+  def buddy_with_deleted
+    buddy || (buddy_id && User.with_deleted { User.find(buddy_id) })
+  end
+
+  def project_with_deleted
+    Project.unscoped { Project.find(stage_with_deleted.project_id) }
+  end
+
+  def stage_with_deleted
+    Stage.unscoped { Stage.find(stage_id) }
+  end
+
+  def user_with_deleted
+    User.unscoped { job.user_id > 0 ? User.find(job.user_id) : nil }
+  end
+
```

Comment:
instead of repeating this logic maybe do a concern like:
```Ruby
def with_deleted(association)
  @with_deleted ||= {}
  @with_deleted.fetch(object) do
    found = 
    @with_deleted[object] = send(object) || (id = send("#{object}_id") && association(object).klass.unscoped.find_by_id(id) }
end
```